src/wp-includes/class-wp-scripts.php

```diff
@@ -455,19 +476,123 @@ public function add_inline_script( $handle, $data, $position = 'after' ) {
 	 * @return string|false Script on success, false otherwise.
 	 */
 	public function print_inline_script( $handle, $position = 'after', $display = true ) {
-		$output = $this->get_data( $handle, $position );
-
+		_deprecated_function( __METHOD__, '6.3.0', 'WP_Scripts::get_inline_script_data() or WP_Scripts::get_inline_script_tag()' );
+		if ( $display ) {
+			$output = $this->get_inline_script_tag( $handle, $position );
+			echo $output;
```

Comment:
```suggestion
			echo $this->get_inline_script_tag( $handle, $position );
```

Reply:
This suggestion won't work because for legacy reasons, the `print_inline_script()` method has to the output:

https://github.com/WordPress/wordpress-develop/blob/12f854cacdea75b244fbd4b5820a508a16c10578/src/wp-includes/class-wp-scripts.php#L457-L471

Nevertheless, when re-looking at this code I see my implementation is wrong. It needs to return the script _data_ not the script _tag_.